src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -0,0 +1,205 @@
+            final boolean logRequestBody = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_LOG_REQUEST_BODY, true);
+            final boolean resolveIndices = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_INDICES, true);
+            final boolean excludeSensitiveHeaders = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_EXCLUDE_SENSITIVE_HEADERS, true);
+
+            final EnumSet<AuditCategory> disabledRestCategories = AuditCategory.parse(getSettingAsList(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES,
+                    DEFAULT_DISABLED_CATEGORIES,
+                    true));
+
+            final EnumSet<AuditCategory> disabledTransportCategories = AuditCategory.parse(getSettingAsList(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_TRANSPORT_CATEGORIES,
+                    DEFAULT_DISABLED_CATEGORIES,
+                    true));
+
+            final Set<String> ignoredAuditUsers = ImmutableSet.copyOf(getSettingAsList(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_USERS,
+                    DEFAULT_IGNORED_USERS,
+                    false));
+
+            final Set<String> ignoredComplianceUsersForRead = ImmutableSet.copyOf(getSettingAsList(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_IGNORE_USERS,
+                    DEFAULT_IGNORED_USERS,
+                    false));
+
+            final Set<String> ignoredComplianceUsersForWrite = ImmutableSet.copyOf(getSettingAsList(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_IGNORE_USERS,
+                    DEFAULT_IGNORED_USERS,
+                    false));
+
+            final Set<String> ignoreAuditRequests = ImmutableSet.copyOf(settings.getAsList(
+                    ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_REQUESTS,
+                    Collections.emptyList()));
+
+            return new Filter(isRestApiAuditEnabled,
+                    isTransportAuditEnabled,
+                    resolveBulkRequests,
+                    logRequestBody,
+                    resolveIndices,
+                    excludeSensitiveHeaders,
+                    ignoredAuditUsers,
+                    ignoredComplianceUsersForRead,
+                    ignoredComplianceUsersForWrite,
+                    ignoreAuditRequests,
+                    disabledRestCategories,
+                    disabledTransportCategories);
+        }
+
+        private static List<String> getSettingAsList(final Settings settings, final String key, final List<String> defaultList, final boolean ignoreCaseForNone) {
```

Comment:
Sounds good!

Reply:
Converting the output to a set has an additional cost when it is passed to `AuditCategory.parse()` as it will be constructing a different set and checking for uniqueness again (not a big deal).